### PR TITLE
setup nucleus cli 

### PR DESCRIPTION
**Pull request by alishakawaguchi**
used this repo as template 
https://github.com/actions/typescript-action

downloads nucleus cli
logs into nucleus is client id and client secret present
end of job clean up logs out of nucleus

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: setup-nucleus-cli
+description: Setup Nucleus CLI, on GitHub Actions runners
+inputs:
+  version:
+    description: Version of Nucleus CLI to install
+    required: false
+    default: 0.0.26
```
Comment (nickzelei) [inline action.yml:7]: if we add the latest, we can just set this to `default: latest` - then we will almost never have to update this GH action.

**Review by nickzelei — COMMENTED** [review]
Looks like the dist folder got checked in. 
Can that be removed and added to the gitignore?

I'm guessing making the version optional will be a little tricky as that will result in having to make a request to the repo to get the latest releases to determine the latest version.
We probably also want to make sure that we only get the latest semver version without any suffixes (like v0.0.1-nick1) kind of thing.
Maybe this can be worked as a follow up feature.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: 'build-test'
+on: # rebuild any PRs and main branch changes
+  pull_request:
+  push:
+    # branches:
+    #   - main
+    #   - 'releases/*'
```
[the remark this answers is not in the document]
Comment (alishakawaguchi) [reply .github/workflows/test.yml]: oops forgot to uncomment

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+          npm install
+      - run: |
+          npm run all
+  test: # make sure the action works on a clean machine without building
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./
+        with:
+          version: 0.0.26
```
[the remark this answers is not in the document]
Comment (alishakawaguchi) [reply .github/workflows/test.yml]: added latest support 👍

File: package.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "setup-nucleus-cli-action",
+  "version": "0.0.0",
+  "private": true,
+  "description": "Github action to install nucleus cli",
+  "main": "lib/main.js",
```
[the remark this answers is not in the document]
Comment (alishakawaguchi) [reply package.json:6]: github action recommends using vercel/ncc to compile the code and to check in the dist folder
https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action

**Pull request merged**